### PR TITLE
kamil/fix-ease-to-padding-transition-on-globe-projection

### PR DESCRIPTION
**Pull request by kamil-sienkiewicz-asi**
## Launch Checklist

Fixes https://github.com/maplibre/maplibre-gl-js/issues/5133

Kapture 2024-11-28 at 14 36 15

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by kamil-sienkiewicz-asi** [comment]
@HarelM Hey, soo i've tried to debug this issue that i've created today and what I noticed was that padding wasn't really interpolated on every `easeTo` step as it is in mercator projection. At first glance seems like copying solution from mercator fixed the issue 🤔 . But I'm not familiar with codebase, so if you could let me know if maybe something else need adjustment along with that PR I would be grateful 🙏

**Review by piotrszacinski — APPROVED** [review]

**Comment by HarelM** [comment]
I remember there was an issue with adding padding, but I'm not sure what it was...
I'll see if I can find similar issues related to it.

**Comment by HarelM** [comment]
I think it might be related to the following issue: 
https://github.com/maplibre/maplibre-gl-js/issues/4891

Padding is applied differently.
Can you check if the linked issue is solved/changed/the same?

**Comment by HarelM** [comment]
@NathanMOlson @kubapelc can you guys take a look at this?

**Comment by kamil-sienkiewicz-asi** [comment]
[Kapture 2024-11-28 at 16.17.47.webm](https://github.com/user-attachments/assets/92473bcf-deea-48b2-9c81-eef9153c9327)

@HarelM I don't see a difference between pre.3 and my branch regarding [this issue](https://github.com/maplibre/maplibre-gl-js/issues/4891) that you've linked. 
To add to that, I think padding is applied correctly (see debug padding), but position to which map should center isn't. Assumption is that consecutive calls of `flyTo` takes into consideration previous padding(?) which would explain why `flyTo` to California ends up near Hawaii. And if I go even further by flying to both of those positions over and over again without moving map it's shifting to the left and bottom on every call and eventually breaks and lands somewhere in Asia.

**Comment by kamil-sienkiewicz-asi** [comment]
@HarelM Hey, I think it's ready to review, not sure why codecov/project is shouting at me tho 🤔

File: src/geo/projection/globe_camera_helper.ts [hunk 1]
```diff
@@ -288,9 +289,7 @@ export class GlobeCameraHelper implements ICameraHelper {
 
         const clonedTr = tr.clone();
         clonedTr.setCenter(constrainedCenter);
-        if (doPadding) {
-            clonedTr.setPadding(options.padding);
-        }
+
```
Comment (HarelM) [inline src/geo/projection/globe_camera_helper.ts:292]: Is "doPadding" variable still needed?

Comment (HarelM) [reply src/geo/projection/globe_camera_helper.ts:292]: It is later, nevermind...

**Comment by HarelM** [comment]
My main concern about this code which I think I recently had a look at is the following comment: 
https://github.com/maplibre/maplibre-gl-js/blob/e25a8d5b557edf9c78f76d4837b9ad1f6380296e/src/geo/projection/mercator_camera_helper.ts#L172

I'm not sure what to make of it and which scenarios we didn't tests as part of this PR...

**Comment by kamil-sienkiewicz-asi** [comment]
@HarelM ah I see why that was needed on mercator, unfortunately I don't have enough knowledge about codebase yet to say something valuable why this works on globe out of the box, I mainly relied on tests, but I could add some more tests if you can figure out some scenarios where it would break 👌

**Comment by HarelM** [comment]
If I'm looking at the mercator code I see some references to offset that can be used (although I'm not sure I understand what's the difference between offset and padding).
I'm also not very knowledgeable in this area of the code, but I would expect some complexities when setting both for globe, can you see if you can find scenarios where this might work differently between mercator and globe?

**Comment by kamil-sienkiewicz-asi** [comment]
[Kapture 2024-11-29 at 16.54.33.webm](https://github.com/user-attachments/assets/19526be9-4791-4d55-8bc2-42eb9d73665e)

Mercator and globe with padding and offset are behaving the same. LngLat values that you see printed out in are unique map.getCenter values (unique to 8th decimal place) - so it doesn't behave weirdly on every consecutive call. I've also tested pitch, roll, elevation, offset, padding and bearing together, and seems like these combinations yield same `map.getCenter()` (to some decimal place, I assume that its a approximation error) in both projections

@HarelM

**Review by HarelM — APPROVED** [review]
Let's hope this won't create other issues, thanks!!

**Pull request merged**